desc/geometry/curve.py

```diff
@@ -806,28 +806,41 @@ def from_values(cls, coords, N=10, basis="xyz", name=""):
 
         # center
         center = np.mean(coords, axis=0)
-        coords = coords - center  # shift to origin
+        coords_centered = coords - center  # shift to origin
 
         # normal
-        U, _, _ = np.linalg.svd(coords.T)
+        U, _, _ = np.linalg.svd(coords_centered.T)
         normal = U[:, -1].T  # left singular vector of the least singular value
 
         # axis and angle of rotation
         Z_axis = np.array([0, 0, 1])
         axis = np.cross(Z_axis, normal)
         angle = np.arccos(np.dot(Z_axis, normal))
         rotmat = rotation_matrix(axis, angle)
-        coords = coords @ rotmat  # rotate to X-Y plane
+        coords_rotated = coords_centered @ rotmat  # rotate to X-Y plane
 
         warnif(
-            np.max(np.abs(coords[:, 2])) > 1e-14,  # check that Z=0 for all points
+            np.max(np.abs(coords_rotated[:, 2])) > 1e-14,  # check that Z=0 for all points
             UserWarning,
             "Curve values are not planar! Using the projection onto a plane.",
         )
 
+        # polar angle
+        s = np.arctan2(coords_rotated[:, 1], coords_rotated[:, 0])
+        unwrapped_s = np.unwrap(s)
+        curve_sign = np.sign(unwrapped_s[-1] - unwrapped_s[0])
+
+        if curve_sign == -1: #original curve was parameterized "backwards" (clockwise) compared to FourierPlanarCurve assumption
+            normal = -normal # flip normal vector direction
+            axis = np.cross(Z_axis, normal)
+            angle = np.arccos(np.dot(Z_axis, normal))
+
+            rotmat = rotation_matrix(axis, angle)
+            coords_rotated = coords_centered @ rotmat  # rotate to X-Y plane
```

Comment:
Why do we need the additional rotation and not just flipping the normal vector direction? I think we only need to flip the normal vector, since that would effectively reverse the parameterization in real space

Reply:
Alternatively, I think we could flip the sign of the Fourier coefficients `r_n` where $m < 0$, since $\cos(-\theta)=\cos(\theta)$ and $\sin(-\theta)=-\sin(\theta)$

Reply:
The weird thing is that the curve points test passes with just flipping the sign of the normal, but the fields are different between spline and normal-flipped-FourierPlanar _unless_ I redo the rotation matrix with the flipped normal. I'll look into this a bit more.

Reply:
So I looked at the coil shapes with and without re-rotating, and it appears that if you only flip the sign of the normal without rotating as well, you mirror the curve as well -- this is because you swap the chirality of the curve. If you drew the curve on a piece of paper, flipping the normal would be looking at the sheet of paper from the back, rather than flipping the piece of paper itself.

Reply:
For some reason this doesn't work, no matter what it seems to need both an axis reversal and a rotation (pancake flip) to get the curves to line up again -- I clarified in the code what the rotation is now doing, but I couldn't find the best way to do it without another rotation.